### PR TITLE
Removing redundant data

### PR DESCRIPTION
**Pull request by akalburgi**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 1]
```diff
@@ -74,8 +74,13 @@ startFromExpr seen world e = case e of
 startFromChoice :: LF.World -> LF.TemplateChoice -> Set.Set Action
 startFromChoice world chc = startFromExpr Set.empty world (LF.chcUpdate chc)
 
+-- We adding template name to archive as we need to have unique choice names
+archiveChoiceAndAction :: LF.Template -> ChoiceAndAction
+archiveChoiceAndAction tpl = ChoiceAndAction (LF.ChoiceName $ tplName tpl <> "_Archive") True Set.empty
+
 templatePossibleUpdates :: LF.World -> LF.Template -> [ChoiceAndAction]
-templatePossibleUpdates world tpl = map (\c -> ChoiceAndAction tpl c (startFromChoice world c)) (NM.toList (LF.tplChoices tpl))
+templatePossibleUpdates world tpl = actions ++ [archiveChoiceAndAction tpl]
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Visual.hs]: Will we not end up with the `Archive` choice being represented twice, once in `actions` and once in `archiveChoiceAndAction`?.

File: compiler/damlc/tests/visual/Basic.dot [hunk 2]
```diff
@@ -2,30 +2,34 @@ digraph G {
 compound=true;
 rankdir=LR;
 subgraph cluster_Group{
-n1[label=Group_Archive][color=red]; 
+n10[label=Archive][color=red];
```
Comment (cocreature) [inline compiler/damlc/tests/visual/Basic.dot]: This seems to confirm my suspicion that we now have `Archive` being represented twice.

Comment (akalburgi) [reply compiler/damlc/tests/visual/Basic.dot]: Ah good catch. will fix it thanks

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 3]
```diff
@@ -26,8 +26,8 @@ data Action = ACreate (LF.Qualified LF.TypeConName)
             | AExercise (LF.Qualified LF.TypeConName) LF.ChoiceName deriving (Eq, Ord, Show )
 
 data ChoiceAndAction = ChoiceAndAction
-    { choiceForTemplate :: LF.Template
-    , choice :: LF.TemplateChoice
+    { choiceName :: LF.ChoiceName
+    , consumingChc :: IsConsuming
```
Comment (rohanjr) [inline compiler/damlc/lib/DA/Cli/Visual.hs]: I would get rid of `Chc` or maybe use `choiceConsuming`.

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 4]
```diff
@@ -26,8 +26,8 @@ data Action = ACreate (LF.Qualified LF.TypeConName)
             | AExercise (LF.Qualified LF.TypeConName) LF.ChoiceName deriving (Eq, Ord, Show )
 
 data ChoiceAndAction = ChoiceAndAction
```
Comment (rohanjr) [inline compiler/damlc/lib/DA/Cli/Visual.hs]: `ChoiceAndActions` (plural)?

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 5]
```diff
@@ -94,17 +99,14 @@ tplName :: LF.Template -> T.Text
...
-            | otherwise = LF.chcName choice
+extractChoiceData (ChoiceAndAction choiceN consuming _) = (choiceN, consuming)
 
 
--- Making choiceName is very weird
 templateWithCreateChoice :: TemplateChoices -> [(LF.ChoiceName, IsConsuming)]
 templateWithCreateChoice TemplateChoices {..} = createChoice : map extractChoiceData choiceAndAction
     where createChoice = (LF.ChoiceName $ tplName template <> "_Create", False)
 
+-- We are adding create as a choice to the graph ,
```
Comment (rohanjr) [inline compiler/damlc/lib/DA/Cli/Visual.hs]:
```suggestion
-- Adding create as a choice to the graph
```

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 6]
```diff
@@ -125,14 +127,17 @@ constructSubgraphsWithLables lookupData tpla@TemplateChoices {..} = SubGraph nod
         nodes = map (\(chc, _) -> (chc, nodeIdForChoice lookupData chc)) choicesInTemplete
         nodesWithCreate = nodes ++ [addCreateChoice tpla lookupData]
 
-actionToChoice :: LF.Template -> Action -> LF.ChoiceName
-actionToChoice tpl (ACreate _) = LF.ChoiceName $ tplName tpl <> "_Create"
-actionToChoice tpl (AExercise _ (LF.ChoiceName "Archive")) = LF.ChoiceName $ tplName tpl <> "_Archive"
-actionToChoice _tpl (AExercise _ chc) = chc
+tplNamet :: LF.TypeConName -> T.Text
```
Comment (rohanjr) [inline compiler/damlc/lib/DA/Cli/Visual.hs]: You're really unqualifying the name right? So maybe `tplNameUnqual`?

File: compiler/damlc/lib/DA/Cli/Visual.hs [hunk 7]
```diff
@@ -125,14 +127,17 @@ constructSubgraphsWithLables lookupData tpla@TemplateChoices {..} = SubGraph nod
...
+
+actionToChoice :: Action -> LF.ChoiceName
+actionToChoice (ACreate LF.Qualified {..}) = LF.ChoiceName $ tplNamet qualObject <> "_Create"
+actionToChoice (AExercise LF.Qualified {..} (LF.ChoiceName "Archive")) = LF.ChoiceName $ tplNamet qualObject <> "_Archive"
+actionToChoice (AExercise _ chc) = chc
 
 choiceActionToChoicePairs :: ChoiceAndAction -> [(LF.ChoiceName, LF.ChoiceName)]
 choiceActionToChoicePairs cha@ChoiceAndAction {..} = pairs
-    where pairs = map (\ac -> (fst $ extractChoiceData cha, actionToChoice choiceForTemplate ac)) (Set.elems actions)
+    where pairs = map (\ac -> (fst $ extractChoiceData cha, actionToChoice ac)) (Set.elems actions)
```
Comment (rohanjr) [inline compiler/damlc/lib/DA/Cli/Visual.hs]: Don't think you need the `where` clause if this is the entire implementation

File: compiler/damlc/lib/DA/Cli/Visual.hs (hunk 3 above)
Comment (akalburgi) [reply compiler/damlc/lib/DA/Cli/Visual.hs]: cool

File: compiler/damlc/lib/DA/Cli/Visual.hs (hunk 6 above)
Comment (akalburgi) [reply compiler/damlc/lib/DA/Cli/Visual.hs]: that sounds a lot better. Done

File: compiler/damlc/lib/DA/Cli/Visual.hs (hunk 7 above)
Comment (akalburgi) [reply compiler/damlc/lib/DA/Cli/Visual.hs]: Done

**Pull request merged**